packages/thumbprint-react/components/Popover/index.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Manager, Reference, Popper } from 'react-popper';
+import startsWith from 'lodash/startsWith';
+
+import * as tokens from '@thumbtack/thumbprint-tokens';
+
+import canUseDOM from '../../utils/can-use-dom';
+
+import { TextButton } from '../Button/index.jsx';
+import { Themed } from '../UIAction/index.jsx';
+import { NavigationCloseTiny } from '../../icons/index.jsx';
+
+import styles from './index.module.scss';
+
+const ESC_KEY = 27;
+
+// === Internal component ===
+
+// Internal component only. Proptypes are defined for the main component `Popover` at the end of the
+// file.
+// eslint-disable-next-line react/prop-types
+const PopoverContent = ({ position, isOpen, children, onCloseClick }) => (
+    <Popper
+        placement={position}
+        modifiers={{
+            offset: { offset: `0, ${tokens.tpSpace3}` },
+            preventOverflow: { boundariesElement: 'window' },
+        }}
+        positionFixed={false}
+    >
+        {({ ref, style, placement, arrowProps }) => (
+            <div
+                ref={ref}
+                className={classNames({
+                    [styles.root]: true,
+                    [styles.open]: isOpen,
+                })}
+                style={style}
+                data-placement={placement}
+            >
+                {children}
+
+                <div className={styles.closeButton}>
+                    <TextButton
+                        accessibilityLabel="Close popover"
+                        iconLeft={<NavigationCloseTiny className={styles.closeButtonIcon} />}
+                        theme="inherit"
+                        onClick={onCloseClick}
+                    />
+                </div>
+
+                <div
+                    className={classNames({
+                        [styles.nubbin]: true,
+                        [styles.nubbinTop]: startsWith(placement, 'bottom'),
+                        [styles.nubbinBottom]: startsWith(placement, 'top'),
+                        [styles.nubbinLeft]: startsWith(placement, 'right'),
+                        [styles.nubbinRight]: startsWith(placement, 'left'),
+                    })}
+                    ref={arrowProps.ref}
+                    style={arrowProps.style}
+                />
+            </div>
+        )}
+    </Popper>
+);
+
+// === Main export ===
+
+const Popover = ({ children, launcher, position, isOpen, onCloseClick, container }) => {
+    // Appends the tooltip right before `</body>` when true.
+    const shouldDisplace = container === 'body';
+
+    const handleKeyUp = event => {
+        if (event.keyCode === ESC_KEY) {
+            onCloseClick();
+        }
+    };
+
+    useEffect(() => {
```

Comment:
We only care about running this on mount and unmount, right? If so, we should add `[]` as the second argument to `useEffect`.